olp-cpp-sdk-core/src/client/OlpClient.cpp

```diff
@@ -239,6 +258,11 @@ CancellationToken OlpClient::CallApi(
     const std::shared_ptr<std::vector<unsigned char>>& post_body,
     const std::string& content_type,
     const NetworkAsyncCallback& callback) const {
+  if (settings_.retry_settings.max_attempts == 0) {
+    callback(HttpResponse(static_cast<int>(http::ErrorCode::CANCELLED_ERROR),
+                          "Operation Cancelled. Max attempts is ZERO."));
+    return CancellationToken();
+  }
```

Comment:
Should be removed, 0 is a valid retry number

Reply:
Fixed